PlayerItem rewinding? fastforwarding? error

I was running received_command.feature:23 `Scenario: Rule supports helper predicates for named commands`

```
bundle exec cucumber features/received_command.feature:23
```

I came across this error:
```
15:22:16.852 [ERROR] [yobj.OpenHAB.DSL.Rules.AutomationRule] - undefined method `fastforward?' for #<Java::OrgOpenhabCoreItemsEvents::ItemCommandEvent:0x5844134d>
Did you mean?  fast_forward? (NoMethodError)
In rule: Execute rule when item received command
/Users/jimmy/Projects/openhab-jruby/tmp/openhab/conf/automation/jsr223/ruby/personal/cucumber_test20211121-93046-1li7ov4.rb:11:in `block in <main>'
org/jruby/RubyArray.java:1865:in `each'
/Users/jimmy/Projects/openhab-jruby/tmp/openhab/conf/automation/jsr223/ruby/personal/cucumber_test20211121-93046-1li7ov4.rb:10:in `block in <main>'
org/jruby/RubyBasicObject.java:2663:in `instance_exec'
/Users/jimmy/Projects/openhab-jruby/tmp/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-4.14.1/lib/openhab/dsl/rules/automation_rule.rb:385:in `process_run_task'
/Users/jimmy/Projects/openhab-jruby/tmp/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-4.14.1/lib/openhab/dsl/rules/automation_rule.rb:330:in `process_task'
/Users/jimmy/Projects/openhab-jruby/tmp/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-4.14.1/lib/openhab/dsl/rules/automation_rule.rb:314:in `process_queue'
/Users/jimmy/Projects/openhab-jruby/tmp/openhab/conf/automation/lib/ruby/gem_home/gems/openhab-scripting-4.14.1/lib/openhab/dsl/rules/automation_rule.rb:70:in `execute'
```

It didn't get caught by the test.

I thought the docs says `rewinding?` and `fast_forwarding?` but only `rewind?` and `fastforward?` is supported. 

I believe `openhab/dsl/types/types.rb` needs fixing. I am just noting this here because I'm in the middle of working on something else.

--- a/features/received_command.feature
+++ b/features/received_command.feature
@@ -20,22 +20,49 @@ Feature:  received_command
     When item "Alarm_Mode" state is changed to "7"
     Then It should log 'Item received command: 7' within 5 seconds
 
-  Scenario: Rule supports helper predicates for named commands
-    Given a deployed rule:
+  Scenario Outline: Rule supports helper predicates for named commands
+    Given items:
+      | type        | name     |
+      | <item_type> | TestItem |
+    And a deployed rule:
       """
       rule 'Execute rule when item received command' do
-        received_command Alarm_Mode
+        received_command TestItem
         run do |event|
-          %i[refresh? on? off? increase? decrease? up? down? stop? move? play? pause? rewind? fastforward? next? previous?].each do |pred|
-            logger.info("Item received #{pred[0..-2]}: #{event.send(pred)}")
-          end
+          predicates = %i[refresh? on? off? increase? decrease? up? down? stop? move? play? pause? rewind? fast_forward? next? previous?]
+
+          predicate_to_command_map = predicates.map {|pred| [pred, pred[0..-2].upcase] }.to_h.merge(fast_forward?: 'FASTFORWARD')
+
+          command = "<command>"
+          predicate = predicate_to_command_map.key(command)
+
+          logger.info("Item received <command>: #{event.send(predicate)}")
+
+          other_results = predicates.grep_v(predicate).select { |other_pred| event.send(other_pred) }
+          logger.info("Other predicates matched: #{other_results.any?} #{other_results}")
         end
       end
       """
-    When item "Alarm_Mode" state is changed to "REFRESH"
-    Then It should log 'Item received up: false' within 5 seconds
-    And It should log 'Item received play: false' within 5 seconds
-    And It should log 'Item received refresh: true' within 5 seconds
+    When item "TestItem" state is changed to "<command>"
+    Then It should log 'Item received <command>: true' within 5 seconds
+    And It should log 'Other predicates matched: false' within 5 seconds
+    Examples:
+      | item_type     | command     |
+      | Switch        | REFRESH     |
+      | Switch        | ON          |
+      | Switch        | OFF         |
+      | Dimmer        | INCREASE    |
+      | Dimmer        | DECREASE    |
+      | Rollershutter | UP          |
+      | Rollershutter | DOWN        |
+      | Rollershutter | STOP        |
+      | Rollershutter | MOVE        |
+      | Player        | PLAY        |
+      | Player        | PAUSE       |
+      | Player        | REWIND      |
+      | Player        | FASTFORWARD |
+      | Player        | NEXT        |
+      | Player        | PREVIOUS    |
 
   Scenario Outline: Rule supports received_command with specific values
     Given a deployed rule:

--- a/lib/openhab/dsl/types/type.rb
+++ b/lib/openhab/dsl/types/type.rb
@@ -27,7 +27,7 @@ module OpenHAB
         #
         def coerce(other)
           logger.trace("Coercing #{self} (#{self.class}) as a request from #{other.class}")
-          return [other.as(self.class), self] if other.is_a?(Type)
+          return [other.as(self.class), self] if other.is_a?(Type) && other.respond_to?(:as)
         end
 
         #

